[LibWebRTC] Consolidate export files and add testing exports

ddkilzer opened the pull request:
#### 161a1abe6d0775b2fbf001f328aabdeea38cf19c
<pre>
[LibWebRTC] Consolidate export files and add testing exports
<a href="https://bugs.webkit.org/show_bug.cgi?id=256560">https://bugs.webkit.org/show_bug.cgi?id=256560</a>
&lt;rdar://109122945&gt;

Reviewed by Youenn Fablet.

General clean-up of libwebrtc export files.  Adds the ability to
export symbols only for testing by introducing a build phase
script in Xcode.

* Source/ThirdParty/libwebrtc/Configurations/DebugRelease.xcconfig:
(WK_EXPORT_SYMBOLS_FOR_TESTING): Add.
- Setting this Xcode variable enables the symbols in
  libwebrtc.testing.exp to be exported when building Debug or
  Release configurations.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.exp: Renamed from Source/ThirdParty/libwebrtc/Configurations/libwebrtc.mac.exp.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.iOS.exp: Remove.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.iOSsim.exp: Remove.
- The symbols in all three files were identical, so consolidated
  them into a single export file.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.testing.exp: Add.
- List of symbols to export when testing.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:
(EXPORTED_SYMBOLS_FILE):
- Update to point to generate export file created by &quot;Generate
  Export Files&quot; build phase script.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc2.exp: Remove.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc4.exp: Remove.
- Remove unused export symbol dumps.

* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
- Rename libwebrtc.mac.exp to libwebrtc.exp, and remove deleted
  files.
(Generate Export Files): Add.
- Add build phase script to generate libwebrtc export file.

Canonical link: <a href="https://commits.webkit.org/263962@main">https://commits.webkit.org/263962@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/af8e7c8783a05649d64267215a7e9e7c6be42f23

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6133 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6319 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6501 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7697 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6478 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6538 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6271 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/9355 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6243 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6263 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5548 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7759 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3738 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5528 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/13432 "7 flakes 1 missing results 144 failures") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5603 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5606 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/7841 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6079 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4975 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5498 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9642 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/734 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5866 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/ThirdParty/libwebrtc/Configurations/DebugRelease.xcconfig:
```diff
@@ -40,6 +40,8 @@ VALIDATE_DEPENDENCIES_INTERNAL_ = $(VALIDATE_DEPENDENCIES_NATIVE_TARGET_$(WK_NOT
 VALIDATE_DEPENDENCIES_INTERNAL_YES = $(WK_VALIDATE_DEPENDENCIES);
 VALIDATE_DEPENDENCIES_NATIVE_TARGET_YES = $(WK_VALIDATE_DEPENDENCIES);
 
+WK_BUILD_FOR_TESTING = YES;
```
I don't think that I agree with the idea of DebugRelease being equal to "build for testing". That's a different concept in Xcode, so we'd be overloading the term here.

ddkilzer replied:
This matches what we already do in WebKitLegacy.

Should I just use the Configuration to enable this mode with a different Xcode variable name that doesn’t include “testing”?

Also, do we actually run tests on any builds other than Debug And Release configs?

ddkilzer replied:
I changed this to `WK_EXPORT_SYMBOLS_FOR_TESTING` since there was no specific feedback given about a different name.  I will assume this is okay unless you say otherwise.

Commit:
[LibWebRTC] Consolidate export files and add testing exports
https://bugs.webkit.org/show_bug.cgi?id=256560
<rdar://109122945>

Reviewed by Youenn Fablet.

General clean-up of libwebrtc export files.  Adds the ability to
export symbols only for testing by introducing a build phase
script in Xcode.

* Source/ThirdParty/libwebrtc/Configurations/DebugRelease.xcconfig:
(WK_EXPORT_SYMBOLS_FOR_TESTING): Add.
- Setting this Xcode variable enables the symbols in
  libwebrtc.testing.exp to be exported when building Debug or
  Release configurations.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.exp: Renamed from Source/ThirdParty/libwebrtc/Configurations/libwebrtc.mac.exp.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.iOS.exp: Remove.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.iOSsim.exp: Remove.
- The symbols in all three files were identical, so consolidated
  them into a single export file.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.testing.exp: Add.
- List of symbols to export when testing.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.xcconfig:
(EXPORTED_SYMBOLS_FILE):
- Update to point to generate export file created by "Generate
  Export Files" build phase script.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc2.exp: Remove.
* Source/ThirdParty/libwebrtc/Configurations/libwebrtc4.exp: Remove.
- Remove unused export symbol dumps.

* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
- Rename libwebrtc.mac.exp to libwebrtc.exp, and remove deleted
  files.
(Generate Export Files): Add.
- Add build phase script to generate libwebrtc export file.

Canonical link: https://commits.webkit.org/263962@main
